forem/forem
# Member Index View Actions: "Assign Role"

## What type of PR is this? (check all applicable)


## Description
This PR adds an "Assign role" modal to the member's action panel in the Member Index View by leveraging the reusable modal work that @aitchiss did. ✨ Additionally, this PR adds e2e tests behind the `member_index_view` feature flag that test that the modal shows as expected.

_Note:_ I accidentally did all of this work under the wrong branch name, so please disregard the branch name and referenced issue number in the branch name! 😴 

## Related Tickets & Documents
- Closes #17480 

## QA Instructions, Screenshots, Recordings
**To QA this PR, please first ensure that you have the `member_index_view` feature flag enabled, and then navigate to `/admin/member_manager/users`.**

Next, test that you are able to open the "Assign role" modal, located within the vertical menu in the righthand corner of each member, by clicking the "Assign role" button within the menu. After, test that you are able to assign a user a role as expected and that upon assigning the role, you are redirected to the view from which you assigned the role, as expected (either `/admin/member_manager/users` or `/admin/member_manager/users/user_id`). This should work for both large and small screens.

#### After:
![Screen Shot 2022-06-08 at 4 04 54 PM](https://user-images.githubusercontent.com/32834804/172725417-e122453d-6a30-4697-89e8-c78931bb9cc5.png)

### UI accessibility concerns?
There shouldn't be. 

## Added/updated tests?

- [x] Yes, however the tests that I did add did not test the functionality of the modal, since that is already covered in `manageRoles.spec.js` and the modal used here is the exact same one, with the exact same functionality. 
      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?
None.

## [optional] What gif best describes this PR or how it makes you feel?

![Michelle Obama saying "I do take the role seriously"](https://media.giphy.com/media/l2JhNgTa7rxzBRlvy/giphy.gif)
